File: app/src/main/java/com/github/ykiselev/playground/init/ScheduleTask.java
package com.github.ykiselev.playground.init;

import com.github.ykiselev.spi.services.schedule.Schedule;

public final class ScheduleTask extends FrameTask {

    private final Schedule schedule;

    public ScheduleTask(Runnable next, Schedule schedule) {
        super(next);
        this.schedule = schedule;
    }

    @Override
    protected void onFrameStart() {
        // no-op
    }

    @Override
    protected void onFrameEnd() {
        schedule.processPendingTasks(2);
    }
}

File: app/src/main/kotlin/com/github/ykiselev/playground/init/ScheduleTask.kt
package com.github.ykiselev.playground.init

import com.github.ykiselev.spi.services.schedule.Schedule

class ScheduleTask(next: Runnable?, private val schedule: Schedule) : FrameTask(next) {

    override fun onFrameEnd() {
        schedule.processPendingTasks(2)
    }
}
